### PR TITLE
cmake: fix compile warnings for clang-cl

### PR DESCRIPTION
**Pull request by zjyhjqs**
[clang-cl is an alternative command-line interface to Clang, designed for compatibility with the Visual C++ compiler, cl.exe.](https://clang.llvm.org/docs/UsersManual.html#clang-cl)

The way to test clang-cl in CMake:
- `CMAKE_<LANGUAGE>_COMPILER_ID`: "Clang"
- `CMAKE_<LANGUAGE>_COMPILER_FRONTEND_VARIANT`: "MSVC"

Note: [CMAKE_\<LANGUAGE\>_COMPILER_FRONTEND_VARIANT](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_FRONTEND_VARIANT.html)) was introduced since CMake 3.14, but the variable [MSVC](https://cmake.org/cmake/help/latest/variable/MSVC.html) works fine here.

---

To pass the GNU style compile options to cl frontend, prefix `/clang:` or `-Xclang` is required to be prepended.

`COMPILE_OPTIONS` here is easier to be transformed, because it is a list, not like `CMAKE_<LANG>_FLAGS` is just a string.

### CODE REVIEW

**Comment by dfandrich** [comment]
Analysis of PR #15337 at [0019d285](https://github.com/curl/curl/commit/0019d285648e54c62ad779069274866ffaae6fe5):

[Test 987 failed](https://github.com/curl/curl/actions/runs/11414285580/job/31762997369#check-step-13), which has NOT been flaky recently, so **there could be a real issue in this PR**.  Note that this test has failed in 2 different CI jobs (the link just goes to one of them).

###### Generated by [Testclutch](https://testclutch.curl.se/)

File: CMake/PickyWarnings.cmake [hunk 1]
```diff
@@ -233,13 +233,23 @@ if(PICKY_COMPILER)
...
+  foreach(_ccopt IN LISTS WPICKY)
+    list(APPEND _wpicky "/clang:${_ccopt}")
+  endforeach()
+  set(WPICKY ${_wpicky})
+endif()
+
 if(WPICKY)
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WPICKY}")
+  add_compile_options(${WPICKY})
   message(STATUS "Picky compiler options:${WPICKY}")
```
Comment (vszakats) [inline CMake/PickyWarnings.cmake:254]: This seems to make picky option passed to tests too, which makes them fail.
Also the message shows semicolons.

I wonder if converting to a string and keep using `CMAKE_C_FLAGS` would
fix both?

```cmake
string(REPLACE ";" " " WPICKY "${WPICKY}")
```

Comment (zjyhjqs) [reply CMake/PickyWarnings.cmake:255]: done

Comment (bagder) [reply CMake/PickyWarnings.cmake:255]: > This seems to make picky option passed to tests too, which makes them fail.

What problems do we get in the tests with picky options?

In configure builds we pass on all the picky compiler options to the tests as well without problems.

Comment (vszakats) [reply CMake/PickyWarnings.cmake:255]: I have to admit it was just a super-quick assesment.

Comparing the two runs it seems the first iteration of this PR got `HAVE_CLOCK_GETTIME_MONOTONIC` and `HAVE_CLOCK_GETTIME_MONOTONIC_RAW` _misdetected_. These are not normally detected for macOS, and detecting them caused some macOS jobs to fail building.

Fail run: https://github.com/curl/curl/actions/runs/11414285596/job/31762998471#step:7:384
Normal run: https://github.com/curl/curl/actions/runs/11418068109/job/31771077457#step:7:384

As it turns out the problem was the reverse of my quick assessment: Before this patch picky warning _were_ passed to tests, and the first iteration of this patch (using `add_compile_options()`) stopped doing that, letting these two tests pass.

Fail test: https://github.com/curl/curl/actions/runs/11414285596/job/31762998471#step:8:5205
Normal test: https://github.com/curl/curl/actions/runs/11418068109/job/31771077457#step:8:5284

TL;DR: we do pass picky warnings to tests and the temp problem was not passing them.

Comment (vszakats) [reply CMake/PickyWarnings.cmake:255]: (which makes me think we still have a problem for builds with picky warnings explicitly disabled!)

**Review by vszakats — APPROVED** [review]

**Comment by vszakats** [comment]
Thanks @zjyhjqs!
